src/ds/d400/d400-device.cpp

```diff
@@ -1081,6 +1103,32 @@ namespace librealsense
         //TODO: Implement
     }
 
+    // Check if need change camera name due to number modifications on one device PID.
+    void update_device_name(std::string& device_name, const ds::d400_caps cap)
+    {
+        std::regex old_name_reg("D455");
+
+		switch (cap)
+		{
+		    case ds::d400_caps::CAP_IR_FILTER:
+                device_name += "F"; // Adding "F" to end of device name if it has IR filter.
+                break;
+
+		    case ds::d400_caps::CAP_IP65:
+                device_name = std::regex_replace(device_name, old_name_reg, "D456"); // Change device name from D455 to D456.
+                break;
+
+            default:
+                auto wrong_cap = ds::d400_capabilities_names.find(cap);
```

Comment:
No need for all of this:
```
default:
    throw invalid_value_exception("capability '" + cap + "' is not supported for device name update");
```

Reply:
`cap`  variable is not a string, so I can't connect it to another string.

Reply:
try this:
```
default:
    throw invalid_value_exception("capability '" + d400_capabilities_names.at( cap ) + "' is not supported for device name update");
```

Reply:
Ok, thank you! I will move forward with your suggestion!

Reply:
I will do next:
`ds::d400_capabilities_names.find(cap)->second`
Because `d400_capabilities_names` is a map.

Reply:
`ds::d400_capabilities_names.find(cap)->second` this use can raise a null reference and crash the SW.
map api has `.at()`  which will throw if the input is not inside the map, which is good for us.

Reply:
Yes, you are right. The first time, IDE showed me a syntax error there.
Now it's OK, something flaky.